Push: Allow an empty destination

jwendell opened the pull request:
In this case assume it's the same as source. This makes it
compatible with `docker push`.

Thus `buildah push my_registry/my_image` will behave like
`buildah push my_registry/my_image docker://my_registry/my_image`

jwendell commented:
If you like the idea I can go ahead and update docs. My use case is to be able to replace `docker push` with `buildah push`

rhatdan commented on cmd/buildah/push.go:
```diff
@@ -78,15 +79,23 @@ var (
+	case 1:
+		src = args[0]
+		destSpec = src
+		logrus.Infof("Destination argument not specified, assuming the same as the source: %s", destSpec)
```
Make this logrus.Debugf

rhatdan commented on cmd/buildah/push.go:
```diff
@@ -78,15 +79,23 @@ var (
+	var src, destSpec string
+	switch len(args) {
+	case 0:
+		return errors.New("source and destination image IDs must be specified")
```
Should probably throw an error if the user specifies 3 or more Args.

TomSweeneyRedHat replied:
This error is no longer true as you only need to specify a source with this change.  I'm struggling with verbiage, but maybe something like:

"At least a source image ID must be specified"

I'm struggling with trying to work the thought of the source becoming the destination if not specified, but can't come up with concise verbiage.  Perhaps the above is enough?

rhatdan commented:
This looks ok to me.  Do we need similar changes to podman push?

rhatdan commented:
You also need to sign your PR.

git commit -a --amend -s
git push --force

TomSweeneyRedHat commented:
Along with docs, a test or two would be good too.

rhatdan commented:
@jwendell We would like to get this into the next buildah release scheduled for tomorrow.  Are you able to continue working on it?

jwendell commented:
@rhatdan done, PTAL.

TomSweeneyRedHat commented:
The push tests are breaking, but I'm not sure why.  It doesn't look like it should be due to this change.  Going to try and retest.

TomSweeneyRedHat commented:
bot, retest this please

rhatdan commented:
@jwendell That test expect an error like

buildah push img1 dest1 -q
No options (-q) can be specified after the image or container name

What does your patched version of buildah do?

jwendell commented:
@rhatdan what I am saying is that even on master that test is failing.

rhatdan commented:
Works on my F28 box?

./tests/test_runner.sh push.bats
++ time bats --tap push.bats
1..4
ok 1 push-flags-order-verification
ok 2 push
ok 3 push with manifest type conversion
ok 4 push with imageid

real	0m14.028s
user	0m3.132s
sys	0m1.648s

jwendell commented:
@rhatdan you're right! What happened here was that I switched to master branch and re-run the test, and it also failed, because I didn't rebuild buildah binary. So, I assumed master was broken too :/

Anyway, just fixed the error and pushed again.

rhatdan commented:
@rh-atomic-bot r+
